packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -108,6 +109,14 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
         throw Exception('NetworkImage is an empty file: $resolved');
 
       return decode(bytes);
+    } catch (e) {
+      // Depending on where the exception was thrown, the image cache may not
+      // have had a chance to track the key in the cache at all.
+      // Schedule a microtask to give the cache a chance to add the key.
+      scheduleMicrotask(() {
+        PaintingBinding.instance.imageCache.evict(key);
+      });
+      rethrow;
```

Comment:
Is it possible that the re-thrown exception is caught and the exception handler puts a replacement image synchronously in the cache that would then unintentionally get removed in the next microtask?

Reply:
Depending on where the exception got thrown, the ImageCache could end up overwriting it with the failing image anyway assuming the user code used the same key.  But in general that would be a bad idea - if your image key caused an exception, you shouldn't be caching anything related to that key, because if the exceptional circumstances change later you'll always get back the "exception" image instead of the good one.

Reply:
So I think the answer to your question is: there's probably some weird way to make that happen, but it would be a bad idea anyway.